metis-indexing/metis-indexing-service/src/main/java/eu/europeana/indexing/service/PublishingService.java

```diff
@@ -0,0 +1,390 @@
+                mongoServer.getDatastore().save(
+                        fullBean.getEuropeanaAggregation().getWebResources());
+            }
+            for (AggregationImpl aggr : fullBean.getAggregations()) {
+                if (aggr.getWebResources() != null) {
+                    mongoServer.getDatastore().save(aggr.getWebResources());
+                }
+            }
+        } else {
+            List<ProvidedCHOImpl> pChos = new ArrayList<>();
+            List<ProxyImpl> proxies = new ArrayList<>();
+            List<AggregationImpl> aggregations = new ArrayList<>();
+            for (ProvidedCHOImpl pCho : fullBean.getProvidedCHOs()) {
+            	
+            	ProvidedCHOImpl pCHOImpl = fullBeanDao.searchByAbout(ProvidedCHOImpl.class, pCho.getAbout());
+            	
+            	if (pCHOImpl != null)
+            		pChos.add(new ProvidedChoUpdater().update(pCHOImpl, pCho, mongoServer));           
+            }
+            
+            for (AggregationImpl aggr : fullBean.getAggregations()) {
+            	
+            	AggregationImpl aggImpl = fullBeanDao.searchByAbout(AggregationImpl.class, aggr.getAbout());
+            	
+            	if (aggImpl != null)
+            		aggregations.add(new AggregationUpdater().update(aggImpl, aggr, mongoServer));            		            	
+            }
+            
+            EuropeanaAggregationImpl eurImpl = fullBeanDao.searchByAbout(EuropeanaAggregationImpl.class,
+            		fullBean.getEuropeanaAggregation().getAbout());
+            
+            if (eurImpl != null)
+            	fullBean.setEuropeanaAggregation(new EuropeanaAggregationUpdater().update(eurImpl,
+            			(EuropeanaAggregationImpl) fullBean.getEuropeanaAggregation(),
+            			mongoServer));                   
+
+            for (ProxyImpl prx : fullBean.getProxies()) {
+            	
+            	ProxyImpl proxImpl = fullBeanDao.searchByAbout(ProxyImpl.class, prx.getAbout());
+            	
+            	if (proxImpl != null)
+            		proxies.add(new ProxyUpdater().update(proxImpl, prx, mongoServer));
+            	
+            }
+            
+            fullBean.setProvidedCHOs(pChos);
+            fullBean.setAggregations(aggregations);
+            fullBean.setProxies(proxies);
+        }
+        
+        fullBean.setAgents(agents);
+        fullBean.setPlaces(places);
+        fullBean.setConcepts(concepts);
+        fullBean.setTimespans(timespans);
+        fullBean.setLicenses(licenses);
+        fullBean.setServices(services);
+    }
+    
+    // Modified and appended legacy code
+    public static FullBeanImpl updateFullBean(FullBeanImpl fullBean, FullBeanDao fullBeanDao) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, MongoDBException {
+        saveEdmClasses(fullBean, false, fullBeanDao);
+        
+        IndexingMongoServer mongoServer = new IndexingMongoServer(fullBeanDao.getDS());

```

Comment:
This object is never used but as a wrapper for a mongo datastore. I suggest removing the use of this object, instead work with the datastore directly. Or even better, not work on the datastore but create methods in FullBeanDaon to perform the tasks you want to perform here.